packages/grid/_modules_/grid/components/toolbar/DensityPicker.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import Button from '@material-ui/core/Button';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import { useIcons } from '../../hooks/utils/useIcons';
+import { ApiContext } from '../api-context';
+import { DensityTypes, Density } from '../../models/gridOptions';
+import { useGridSelector } from '../../hooks/features/core/useGridSelector';
+import { optionsSelector } from '../../hooks/utils/useOptionsProp';
+import { DensityOption } from '../../models/api/densityApi';
+import { densitySizeSelector } from '../../hooks/features/density';
+
+export const DensityPicker = React.memo(function DensityPicker() {
```

Comment:
what if we want to render the menu from another button like in the column menu.
This way, it is all coupled together, so we can't

Reply:
> what if

Let's wait to get a request in this direction.

Reply:
Yes that is the reason why I have it both in the same file. IMO it makes more sense to have them coupled together so if devs want to have their own component, when they disable the feature all presentational code related to it will be removed.

Reply:
Agree, I think that we should focus on the composability of the toolbar first.